src/Files.App.Controls/Storage/StorageRing/StorageRing.cs

```diff
@@ -0,0 +1,1025 @@
+		/// <param name="percentage">The percentage to convert.</param>
+		/// <param name="minValue">The minimum value of the input range.</param>
+		/// <param name="maxValue">The maximum value of the input range.</param>
+		/// <returns>The percentage value (between 0 and 100).</returns>
+		private double PercentageToValue(double percentage, double minValue, double maxValue)
+		{
+			double convertedValue = percentage * (maxValue - minValue) / 100.0;
+
+			// Ensure the converted value stays within the specified range
+			if (convertedValue < minValue)
+				convertedValue = minValue;
+			else if (convertedValue > maxValue)
+				convertedValue = maxValue;
+
+			return convertedValue;
+		}
+
+		/// <summary>
+		/// Calculates the total angle needed to accommodate a gap between two strokes around a circle.
+		/// </summary>
+		/// <param name="thickness">The Thickness radius to measure.</param>
+		/// <param name="radius">The radius of the rings.</param>
+		/// <returns>The gap angle (sum of angles for the larger and smaller strokes).</returns>
+		private double GapThicknessToAngle(double radius, double thickness)
+		{
+			if (radius > 0 && thickness > 0)
+			{
+				// Calculate the maximum number of circles
+				double n = Math.PI * (radius / thickness);
+
+				// Calculate the angle between each small circle
+				double angle = 360.0 / n;
+
+				return angle;
+			}
+
+			return 0;
+		}
+
+		/// <summary>
+		/// Calculates an adjusted angle using linear interpolation (lerp) between the start and end angles.
+		/// </summary>
+		/// <param name="startAngle">The initial angle.</param>
+		/// <param name="endAngle">The final angle.</param>
+		/// <param name="valueAngle">A value between 0 and 1 representing the interpolation factor.</param>
+		/// <returns>The adjusted angle based on linear interpolation.</returns>
+		private static double GetInterpolatedAngle(double startAngle, double endAngle, double valueAngle)
+		{
+			// Linear interpolation formula (lerp): GetInterpolatedAngle = (startAngle + valueAngle) * (endAngle - startAngle)
+			return (startAngle + valueAngle) * (endAngle - startAngle);
+		}
+
+		/// <summary>
+		/// Example quadratic ease-in-out function
+		/// </summary>
+		private double EaseInOutFunction(double t)
+		{
+			return t < 0.5 ? 2 * t * t : 1 - Math.Pow(-2 * t + 2, 2) / 2;
+		}
+
+		/// <summary>
+		/// Example ease-out cubic function
+		/// </summary>
+		static double EaseOutCubic(double t)
+		{
+			return 1.0 - Math.Pow(1.0 - t, 3.0);
+		}
+
+		/// <summary>
+		/// Checks True if the Angle range is a Full Circle
+		/// </summary>
+		/// <param name="MinAngle"></param>
+		/// <param name="MaxAngle"></param>
+		/// <returns></returns>
+		public static bool IsFullCircle(double MinAngle, double MaxAngle)
+		{
+			// Calculate the absolute difference between angles
+			double angleDifference = Math.Abs(MaxAngle - MinAngle);
+
+			// Check if the angle difference is equal to 360 degrees
+			return angleDifference == 360;
```

Comment:
Comparing floating point numbers is a complex matter that is heavily discussed in computer science and more often than not, this comparison will not work as you expect. If you really need to be accurate a safer way to compare would be: `return Math.Abs(angleDifference - 360) < Double.Epsilon);` Given we are dealing with pixels and rendering, checking if we are within `1` or `0.1` of 360 should be fine.

Reply:
I will make this change